### PR TITLE
Cleanup isa

### PR DESCRIPTION
**Pull request by ibludau**
Added and updated traces methods.

### CODE REVIEW

File: R/tracesMethods.R [hunk 1]
```diff
@@ -53,36 +96,149 @@ toLongFormat <- function(traces.dt) {
...
+#' # Protein traces
+#' proteinTraces=exampleProteinTraces
+#' plot(proteinTraces)
+#' # Annotate traces with molecular weight to include molecular weight information in plot.
+#' calibrationTable <- exampleCalibrationTable
+#' # Perform molecular weight calibration:
+#' calibration = calibrateSECMW(calibrationTable)
+#' # Perform molecular weight annotation:
+#' mwProteinTraces <- annotateMolecularWeight(proteinTraces, calibration)
+#' plot(mwProteinTraces)
```
Comment (mffrank) [inline R/tracesMethods.R:140]: Maybe add an example where a subset is performed and then plotted.

File: R/tracesMethods.R [hunk 2]
```diff
@@ -53,36 +96,149 @@ toLongFormat <- function(traces.dt) {
...
+#'           Only reported if sibling peptide correlations were previously calculated.
+#'          }
+#' @examples
+#' summary(examplePeptideTracesFiltered)
 #' @export
-summary.traces <- function(traces.obj) {
-  no_traces <- nrow(traces.obj$trace_annotation)
-  no_decoys <- length(grep("DECOY", traces.obj$trace_annotation$protein_id))
+summary.traces <- function(traces) {
+  no_traces <- nrow(traces$trace_annotation)
```
Comment (mffrank) [inline R/tracesMethods.R]: add traces test

File: tests/testthat/test_annotateTraces.R [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                                  replace_whitespace = TRUE)
+
+test_that("Output format",{
+  # test if untouched traces parts stay identical
+  testthat::expect_identical(examplePeptideTracesUnannotated$traces, annotatedTraces$traces)
+  testthat::expect_identical(examplePeptideTracesUnannotated$trace_type, annotatedTraces$trace_type)
+  testthat::expect_identical(examplePeptideTracesUnannotated$fraction_annotation, annotatedTraces$fraction_annotation)
+  testthat::expect_identical(examplePeptideTracesUnannotated$trace_annotation$id, annotatedTraces$trace_annotation$id)
+  # test if annotation was successful
+  testthat::expect_true(all(names(exampleTraceAnnotation) %in% c(names(annotatedTraces$trace_annotation),"Entry")))
```
Comment (mffrank) [inline tests/testthat/test_annotateTraces.R]: add traces test function

File: tests/testthat/test_tracesMethods.R [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  testthat::expect_equal(length(names(peptideSubsettedPeptideTraces$traces)), length(subsetFractions)+1)
+  testthat::expect_equal(peptideSubsettedPeptideTraces$fraction_annotation$id, subsetFractions)
+  ## test proteinTraces subsetting
+  testthat::expect_equal(length(proteinSubsettedProteinTraces$traces$id), length(subsetProtein))
+  testthat::expect_equal(length(proteinSubsettedProteinTraces$trace_annotation$id), length(subsetProtein))
+  ## test if subsetting by thomething other than "id" works
+  testthat::expect_equal(length(proteinSubsettedPeptideTraces$traces$id), 42)
+  testthat::expect_equal(length(proteinSubsettedPeptideTraces$trace_annotation$id), 42)
+  ## test if traces and trace_annotation are identical (no shift in order etc)
+  testthat::expect_identical(proteinSubsettedPeptideTraces$trace_annotation$id,proteinSubsettedPeptideTraces$traces$id)
```
Comment (mffrank) [inline tests/testthat/test_tracesMethods.R:28]: add traces test function:)

**Review by mffrank — CHANGES_REQUESTED** [review]
See comments, but looks good!

**Comment by ibludau** [comment]
All suggestions were incorporated :)

**Review by mffrank — APPROVED** [review]
Nice:)

**Pull request merged**